hastexo/public/js/main.js

```diff
@@ -113,168 +113,166 @@ function HastexoXBlock(runtime, element, configuration) {
           terminal_ws_url = prot_map[location.protocol] + '//' + location.hostname + configuration.terminal_url;
         }
 
-        /* Load app dynamically. */
-        $.cachedScript(terminal_http_url + 'guacamole-common-js/all.min.js').done(function() {
-            terminal_client = new Guacamole.Client(
-                new Guacamole.WebSocketTunnel(terminal_ws_url + "websocket-tunnel")
-            );
-            terminal_element = terminal_client.getDisplay().getElement();
-
-            /* Show the terminal.  */
-            $("#terminal").append(terminal_element);
-
-            /* Disconnect on tab close. */
-            window.onunload = function() {
-                terminal_client.disconnect();
-            };
+        /* Initialize Guacamole Client */
```

Comment:
This piques my curiosity. :) Can you elaborate on why we're changing this behavior? It would seem to me like waiting for the `.done()` event would be a good thing; does it break things in this instance?

Reply:
We previously defined the `guacamole-common-js` plugin in `pom.xml` and were able to load it here dynamically. So now I was able to make it work by adding `guacamole-common-js/all.min.js` to `public/js` and loading the resource in the `student_view` in `hastexo.py`.

Reply:
Sounds good. Thanks for the explanation!